chips/nrf52/src/qdec.rs

```diff
@@ -0,0 +1,278 @@
+            us128 = 0,
+            us256 = 1,
+            us512 = 2,
+            us1024 = 3,
+            us2048 = 4,
+            us4096 = 5,
+            us8192 = 6,
+            us16384 = 7,
+            ms32 = 8,
+            ms65 = 9,
+            ms131 = 10
+        ]
+    ],
+    ReportPer [
+        REPORTPER OFFSET(0) NUMBITS(4) [
+            hz10 = 0,
+            hz40 = 1,
+            hz80 = 2,
+            hz120 = 3,
+            hz160 = 4,
+            hz200 = 5,
+            hz240 = 6,
+            hz280 = 7,
+            hz1 = 8
+        ]
+    ],
+    Acc [
+        ACC OFFSET(0) NUMBITS(32)
+    ],
+    AccDbl [
+        ACCDBL OFFSET(0) NUMBITS(4)
+    ]
+];
+
+/// This defines the beginning of memory which is memory-mapped to the Qdec
+/// This base is declared under the Registers Table 3
+const QDEC_BASE: StaticRef<QdecRegisters> =
+    unsafe { StaticRef::new(0x40012000 as *const QdecRegisters) };
+
+pub static mut QDEC: Qdec = Qdec::new();
+
+#[derive(PartialEq, Eq)]
+enum QdecState {
+    Start,
+    Stop,
+}
+/// Qdec type declaration: gives the Qdec instance registers and a client
+pub struct Qdec {
+    registers: StaticRef<QdecRegisters>,
+    client: OptionalCell<&'static dyn kernel::hil::qdec::QdecClient>,
+    state: QdecState,
+}
+
+/// Qdec impl: provides the Qdec type with vital functionality including:
+impl Qdec {
+    const fn new() -> Qdec {
+        let qdec = Qdec {
+            registers: QDEC_BASE,
+            client: OptionalCell::empty(),
+            state: QdecState::Start,
+        };
+        qdec
+    }
+
+    /// sets pins_a and pins_b to be the output pins for whatever the encoding device is  
+    pub fn set_pins(&self, pin_a: pinmux::Pinmux, pin_b: pinmux::Pinmux) {
+        let regs = self.registers;
+        regs.psel_a.write(
+            PinSelect::Pin.val(pin_a.into()) + PinSelect::Port.val(0) + PinSelect::Connect.val(0),
+        );
+        regs.psel_b.write(
+            PinSelect::Pin.val(pin_b.into()) + PinSelect::Port.val(0) + PinSelect::Connect.val(0),
+        );
+    }
+
+    pub fn set_client(&self, client: &'static dyn kernel::hil::qdec::QdecClient) {
+        self.client.set(client);
+    }
+
+    /// When an interrupt occurs, check to see if any
+    /// of the interrupt register bits are set. If it
+    /// is, then put it in the client's bitmask
+    pub(crate) fn handle_interrupt(&self) {
+        let regs = &*self.registers;
+        self.client.map(|client| {
+            // For each of 4 possible compare events, if it's happened,
+            // clear it and sort its bit in val to pass in callback
+            for i in 0..regs.events_arr.len() {
+                if regs.events_arr[i].is_set(Event::READY) {
+                    regs.events_arr[i].set(0);
+                    match i {
+                        0 => {
+                            client.sample_ready();
+                        }
+                        1 => { /*No handling for REPORTRDY*/ }
```

Comment:
Should this (and 3 below) be `unimplemented!()`, or is this something that will never need to be implemented (if the latter, a bit more exposition in the comment would be helpful)?